### PR TITLE
Fix missing errors after using select_nested_models

### PR DESCRIPTION
**Pull request by QRemy**
Cahche also the covariacne matrix otherwise errors are not restored ( might have the same problem with restore_status).

### CODE REVIEW

**Comment by bkhelifi** [comment]
Can you add the associated test to validate the identified issues?

**Comment by QRemy** [comment]
Tests added  but I did not test for linked models yet but we can go without it for now as usage is limited.

**Comment by QRemy** [comment]
Should be ok for linked params (but the test is probably not the best)

File: gammapy/modeling/selection.py [hunk 1]
```diff
@@ -143,31 +143,38 @@ def run(self, datasets, apply_selection=True):
...
+        cache["values"] = [p.value for p in datasets.models.parameters]
+        cache["error"] = [p.error for p in datasets.models.parameters]
         for p, val in zip(self.parameters, self.null_values):
             if isinstance(val, Parameter):
                 p.__dict__ = val.__dict__
             else:
                 p.value = val
-                p.frozen = True
-        return object_cache, prev_pars
+        p.frozen = True
```
Comment (registerrier) [inline gammapy/modeling/selection.py:163]: is it the correct indentation?

File: gammapy/modeling/tests/test_selection.py [hunk 2]
```diff
@@ -65,12 +69,16 @@ def test_test_statistic_detection_other_frozen(fermi_datasets):
...
     # TODO: better test with simulated data ?
-    model = fermi_datasets.models["Crab Nebula"]
+
+    models = Models.read("$GAMMAPY_DATA/fermi-3fhl-crab/Fermi-LAT-3FHL_models.yaml")
+
+    model = models["Crab Nebula"]
     model2 = model.copy(name="other")
     model2.spectral_model.alpha.value = 2.4
-    fermi_datasets.models = fermi_datasets.models + [model2]
+    model2.spectral_model.alpha.error = 1
```
Comment (registerrier) [inline gammapy/modeling/tests/test_selection.py:79]: why do you set the error manually?

**Review by registerrier — COMMENTED** [review]
Thanks @QRemy . Looks good. I have left two minor comments inline.

File: gammapy/modeling/tests/test_selection.py (hunk 2 above)
Comment (QRemy) [reply gammapy/modeling/tests/test_selection.py:79]: not necessary, removed

File: gammapy/modeling/selection.py (hunk 1 above)
Comment (QRemy) [reply gammapy/modeling/selection.py:163]: fixed

**Pull request merged**